pandas/tools/merge.py

```diff
@@ -1369,7 +1369,9 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
                 clean_keys.append(k)
                 clean_objs.append(v)
             objs = clean_objs
-            keys = clean_keys
+            name = getattr(keys, 'name', None)
+            keys = _ensure_index(Index(clean_keys))
```

Comment:
yes appears that is correct: https://github.com/pydata/pandas/pull/14292/commits/cdc76f6f0e2fd5e307f3a335afc23385d1e8234f looks fine.